docs/getting-started-guides/azure.md

```diff
@@ -0,0 +1,154 @@
+
+```shell
+export AZURE_DEPLOY_ID="" # autogenerated if blank
+export AZURE_LOCATION="westus"
+export AZURE_RESOURCE_GROUP="" # generated from AZURE_DEPLOY_ID if unset
+export AZURE_MASTER_SIZE="Standard_A1"
+export AZURE_NODE_SIZE="Standard_A1"
+export AZURE_USERNAME="kube"
+export NUM_NODES=3
+export AZURE_AUTH_METHOD="device"
+```
+
+
+By default, this will deploy a cluster with 4 `Standard_A1`-sized VMs: one master node, three worker nodes. This process takes about 5 to 10 minutes. Once the cluster is up, connection information to the cluster will be displayed. Additionally, your `kubectl` configuration will be updated to know about this cluster and this new cluster will be set as the active context.
+
+The Azure deployment process produces an output directory `cluster/azure/_deployments/${AZURE_DEPLOY_ID}`. In this directory you will find the PKI and SSH assets created for the cluster, as well as a script named `util.sh`. Here are some examples of its usage:
+
+```
+$ cd cluster/azure/_deployments/kube-20160316-001122/
+
+# This uses the client cert with curl to make an http call to the apiserver.
+$ ./util.sh curl api/v1/nodes
+
+# This uses the client cert with kubectl to target this deployment's apiserver.
+$ ./util.sh kubectl get nodes
+
+# This alters the current kubectl configuration to point at this cluster.
+$ ./util.sh configure-kubectl
+
+# This will deploy the kube-system namespace, the SkyDNS addon, and the kube-dashboard addon.
+$ ./util.sh deploy-addons
+
+# This uses the ssh private key to copy the private key itself to the master node.
+$ ./util.sh copykey
+
+# This uses the ssh private key to open an ssh connection to the master.
+$ ./util.sh ssh
+```
+
+### Cluster deployment examples
+
+#### Using `kubectl proxy`
+
+`kubectl proxy` is currently used to access to access deployed services.
+
+```
+kubectl proxy --port=8001
+```
+
+Deployed services are available at: `http://localhost:8001/api/v1/proxy/namespaces/<namespace>/services/<service_name>`.
+
+
+#### Addon: SkyDNS
+
+You can deploy the [SkyDNS addon](https://github.com/kubernetes/kubernetes/tree/release-1.2/cluster/addons/dns):
+
+```
+kubectl create -f https://raw.githubusercontent.com/colemickens/azkube/v0.0.4/templates/coreos/addons/skydns.yaml
+```
+
+
+#### Addon: Kube-Dashboard
+
+This will deploy the [`kube-dashboard`](https://github.com/kubernetes/dashboard) addon:
+
+```
+kubectl create -f https://raw.githubusercontent.com/colemickens/azkube/v0.0.4/templates/coreos/addons/kube-dashboard.yaml
+```
+
+The dashboard is then available at: `http://localhost:8001/api/v1/proxy/namespaces/kube-system/services/dashboard-canary`.
+
+#### Example: Guestbook
+
+This will deploy the [`guestbook example`](https://github.com/kubernetes/kubernetes/blob/release-1.2/examples/guestbook/README.md) (the all-in-one variant):
```

Comment:
URL should be https://github.com/kubernetes/kubernetes/blob/{{page.githubbranch}}/examples/guestbook/README.md